crates/runtime/src/logic.rs

```diff
@@ -123,6 +125,7 @@ impl<'a> VMLogic<'a> {
             events: vec![],
             actions: vec![],
             root_hash: None,
+            proposals: BTreeMap::new(),
```

Comment:
it still needs to make it out through `Outcome` so you need to tweak `VMLogic::finish`